src/Plugin.php

```diff
@@ -72,6 +72,12 @@ class_exists(__NAMESPACE__ . '\\' . $class);
         $this->config = $composer->getConfig();
         $this->package = $composer->getPackage();
 
+        $cacheDir = rtrim($this->config->get('cache-files-dir'), '\/');
+
+        if (preg_match('{(^|[\\\\/])(\$null|nul|NUL|/dev/null)([\\\\/]|$)}', $cacheDir)) {
```

Comment:
Function `preg_match` returns false on error, so better compare the result to `1`.

Reply:
This is working for me ^^